core/src/main/java/ai/timefold/solver/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -219,226 +219,233 @@ public final class ConstraintCollectors {
      * The default result of the collector (e.g. when never called) is {@code 0}.
      *
      * @param <A> type of the matched fact
-     * @return never null
      */
-    public static <A> UniConstraintCollector<A, ?, Integer> sum(ToIntFunction<? super A> groupValueMapping) {
+    public static <A> @NonNull UniConstraintCollector<A, ?, Integer> sum(@NonNull ToIntFunction<? super A> groupValueMapping) {
         return InnerUniConstraintCollectors.sum(groupValueMapping);
     }
 
     /**
      * As defined by {@link #sum(ToIntFunction)}.
      */
-    public static <A> UniConstraintCollector<A, ?, Long> sumLong(ToLongFunction<? super A> groupValueMapping) {
+    public static <A> @NonNull UniConstraintCollector<A, ?, Long>
+            sumLong(@NonNull ToLongFunction<? super A> groupValueMapping) {
         return InnerUniConstraintCollectors.sum(groupValueMapping);
     }
 
     /**
      * As defined by {@link #sum(ToIntFunction)}.
      */
-    public static <A, Result> UniConstraintCollector<A, ?, Result> sum(Function<? super A, Result> groupValueMapping,
+    // TODO: nullability of further params (suppose all are @NonNull)
```

Comment:
@triceo should `zero`, `adder` and `subtractor` all be `@NonNull` here?

Reply:
Yes. There are overloads for that method which do not require these arguments - so when an argument is actually present, it is required unless Javadoc says otherwise.